sdk/src/asset_handlers/pdf_io.rs

```diff
@@ -0,0 +1,256 @@
+// Copyright 2023 Adobe. All rights reserved.
+// This file is licensed to you under the Apache License,
+// Version 2.0 (http://www.apache.org/licenses/LICENSE-2.0)
+// or the MIT license (http://opensource.org/licenses/MIT),
+// at your option.
+
+// Unless required by applicable law or agreed to in writing,
+// this software is distributed on an "AS IS" BASIS, WITHOUT
+// WARRANTIES OR REPRESENTATIONS OF ANY KIND, either express or
+// implied. See the LICENSE-MIT and LICENSE-APACHE files for the
+// specific language governing permissions and limitations under
+// each license.
+
+use std::{fs::File, path::Path};
+
+use crate::{
+    asset_io::{AssetIO, CAIReader, CAIWriter, HashObjectPositions},
+    utils::pdf_utils::{C2paPdf, Pdf},
+    CAIRead, Error,
+    Error::{JumbfNotFound, NotImplemented, PdfReadError},
+};
+
+static SUPPORTED_TYPES: [&str; 2] = ["pdf", "application/pdf"];
+static WRITE_NOT_IMPLEMENTED: &str = "PDF write functionality will be added in a future release";
+
+pub struct PdfIO {}
+
+impl CAIReader for PdfIO {
+    fn read_cai(&self, asset_reader: &mut dyn CAIRead) -> crate::Result<Vec<u8>> {
+        let mut buffer: Vec<u8> = vec![];
```

Comment:
Long-term, is there a way we can avoid this full copy of the asset?

Reply:
Looks like they support Document::load_from(reader: Read), this can be changed.

Reply:
Updated - I ran into an issue with using mockall and the trait `Read`, but I've removed the constructors from the `C2paPdf` trait, as those aren't very useful for testing anyway, and got around the issue altogether!